src/pages/iou/IOUTransactions.js

```diff
@@ -0,0 +1,87 @@
+import React, {Component} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import ONYXKEYS from '../../ONYXKEYS';
+import ReportActionPropTypes from '../home/report/ReportActionPropTypes';
+import iouTansactionPropTypes from './iouTansactionPropTypes';
+import ReportTransaction from '../../components/ReportTransaction';
+
+const propTypes = {
+    // Actions from the ChatReport
+    reportActions: PropTypes.shape(ReportActionPropTypes),
+
+    // ReportID for the associated chat report
+    chatReportID: PropTypes.number.isRequired,
+
+    // ReportID for the associated IOU report
+    iouReportID: PropTypes.number.isRequired,
+
+    // Transactions for this IOU report
+    transactions: PropTypes.arrayOf(PropTypes.shape(iouTansactionPropTypes)),
+};
+
+const defaultProps = {
+    reportActions: {},
+    transactions: [],
+};
+
+class IOUTransactions extends Component {
+    constructor(props) {
+        super(props);
+
+        this.getActionForTransaction = this.getActionForTransaction.bind(this);
+    }
+
+    /**
+     * Given a transaction from an IOU Report, returns the chatReport action with a matching transactionID. Unless
+     * something has gone wrong with our storing logic, there should always exist an action for each transaction.
+     *
+     * @param {Object} transaction
+     * @returns {Object} action
+     */
+    getActionForTransaction(transaction) {
+        const matchedAction = _.find(this.props.reportActions, (action) => {
+            if (action && action.originalMessage
+                && action.originalMessage.IOUTransactionID === transaction.transactionID) {
+                return action;
+            }
+            return false;
+        });
+        if (!matchedAction) {
+            throw new Error(`Unable to locate a matching report action for transaction ${transaction.transactionID}!`);
```

Comment:
Why is an error being thrown here? Is anything going to catch it and do anything with it? Is it something to be worried about?

Reply:
This will never happen, but it was attempting to clarify the fact that there should always be a match here. 

As above, this logic has [already been replaced](https://github.com/Expensify/Expensify.cash/blob/8d6177a94012dc2d8f0295b52fe45ce9c3e12b5d/src/pages/iou/IOUTransactions.js#L26-L46), so I'm going to resolve it.